indexer/services/ender/src/handlers/handler.ts

```diff
@@ -49,20 +51,23 @@ export abstract class Handler<T> {
   blockEventIndex: number;
   event: T;
   abstract eventType: string;
+  messageReceivedTimestamp?: string;
 
   constructor(
     block: IndexerTendermintBlock,
     blockEventIndex: number,
     indexerTendermintEvent: IndexerTendermintEvent,
     txId: number,
     event: T,
+    messageReceivedTimestamp?: string,
   ) {
     this.block = block;
     this.blockEventIndex = blockEventIndex;
     this.indexerTendermintEvent = indexerTendermintEvent;
     this.timestamp = DateTime.fromJSDate(block.time!);
     this.txId = txId;
     this.event = event;
+    this.messageReceivedTimestamp = messageReceivedTimestamp;
```

Comment:
can't we just use the timestamp from `this.block`? I'm pretty sure we already have a helper to pull timestamp out of it

Reply:
We could, but I wanted to only track Indexer system latency without block gossiping latency.